apps/hash-frontend/src/components/hooks/use-account-pages.ts

```diff
@@ -21,17 +42,57 @@ export const useAccountPages = (
 ): AccountPagesInfo => {
   const { hashInstance } = useHashInstance();
 
-  const { data, loading } = useQuery<
-    GetAccountPagesTreeQuery,
-    GetAccountPagesTreeQueryVariables
-  >(getAccountPagesTree, {
-    variables: { ownedById, includeArchived },
+  const { data, loading, refetch } = useQuery<
+    StructuralQueryEntitiesQuery,
+    StructuralQueryEntitiesQueryVariables
+  >(structuralQueryEntitiesQuery, {
+    variables: getAccountPagesVariables({
+      ownedById,
+      includeArchived,
+    }),
     skip: !ownedById || !hashInstance?.properties.pagesAreEnabled,
   });
 
-  const pages = useMemo(() => {
-    return data?.pages ?? [];
-  }, [data?.pages]);
+  const pages = useMemo<SimplePage[]>(() => {
+    const subgraph = data?.structuralQueryEntities.subgraph;
+
+    if (!subgraph) {
+      return [];
+    }
+
+    const typedSubgraph =
+      mapGqlSubgraphFieldsFragmentToSubgraph<EntityRootType>(subgraph);
+
+    return typedSubgraph.roots.map((root) => {
+      const pageEntityRevisions = typedSubgraph.vertices[root.baseId];
+
+      if (!pageEntityRevisions) {
+        throw new Error(`Could not find page entity with id ${root.baseId}`);
+      }
+
+      const latestPage = typedValues(pageEntityRevisions)[0]!
+        .inner as Entity<PageProperties>;
+
+      const pageOutgoingLinks = getOutgoingLinkAndTargetEntities(
+        subgraph,
+        latestPage.metadata.recordId.entityId,
+      );
+
+      const parentLink = pageOutgoingLinks.find(
+        ({ linkEntity }) =>
+          linkEntity[0]!.metadata.entityTypeId ===
+          systemTypes.linkEntityType.parent.linkEntityTypeId,
+      );
+
+      const parentPage = parentLink?.rightEntity[0] ?? null;
+
+      return {
+        ...simplifyProperties(latestPage.properties),
+        metadata: latestPage.metadata,
+        parentPage: parentPage ? { metadata: parentPage.metadata } : null,
+      };
+    });
```

Comment:
Maybe I'm missing something, but the query doesn't seem to be returning multiple revisions of pages, so wouldn't it suffice to just do this?
```suggestion
    return getRoots(typedSubgraph).map((latestPage) => {
      const pageOutgoingLinks = getOutgoingLinkAndTargetEntities(
        subgraph,
        latestPage.metadata.recordId.entityId,
      );

      const parentLink = pageOutgoingLinks.find(
        ({ linkEntity }) =>
          linkEntity[0]!.metadata.entityTypeId ===
          systemTypes.linkEntityType.parent.linkEntityTypeId,
      );

      const parentPage = parentLink?.rightEntity[0] ?? null;

      return {
        ...simplifyProperties(latestPage.properties as PageProperties),
        metadata: latestPage.metadata,
        parentPage: parentPage ? { metadata: parentPage.metadata } : null,
      };
    });
```

Reply:
Yes, it would – thank you!